pkg/apiserver/profiling/model.go

```diff
@@ -84,11 +100,15 @@ func NewTask(ctx context.Context, taskGroup *TaskGroup, target model.RequestTarg
 }
 
 func (t *Task) run() {
-	fileNameWithoutExt := fmt.Sprintf("profiling_%d_%d_%s", t.TaskGroupID, t.ID, t.Target.FileName())
-	protoFilePath, rawDataType, err := profileAndWritePprof(t.ctx, t.fetchers, &t.Target, fileNameWithoutExt, t.taskGroup.ProfileDurationSecs)
+	fileNameWithoutExt := fmt.Sprintf("profiling_%d_%d_%s_%s", t.TaskGroupID, t.ID, t.ProfilingType, t.Target.FileName())
+	protoFilePath, rawDataType, err := profileAndWritePprof(t.ctx, t.fetchers, &t.Target, fileNameWithoutExt, t.taskGroup.ProfileDurationSecs, t.ProfilingType)
 	if err != nil {
 		t.Error = err.Error()
-		t.State = TaskStateError
+		if err.Error() == "unsupported profiling type" {
```

Comment:
It is not a good practice to test error against its message. You can use errorx to make a typed error and use [`errorx.IsOfType(..)`](https://pkg.go.dev/github.com/joomcode/errorx#IsOfType) to test against it.